### DISCUSSION TITLE
Table not refreshing on update

### DISCUSSION THREAD

**Discussion by etmartinkazoo** [livewire/livewire#6231] [1. Help]
I am running a Laravel 10 app in PHP 8.1 with Livewire 3, beta v7.

I have a table for "Tasks" that looks like this:
<img width="1239" alt="image" src="https://github.com/livewire/livewire/assets/13721164/e3d98da2-ea0f-4051-b271-df899923e609">

The following is my component for the dropdown:
```
<?php

namespace App\Livewire;

use Livewire\Component;
use App\Models\User;

class DropdownFilter extends Component
{
    public $selectedUser = null;

    public $users;

    public function updatedSelectedUser()
    {
        $this->selectedUser = intval($this->selectedUser);
        $this->dispatch('user-changed', $this->selectedUser);
    }

    public function mount()
    {
        $this->selectedUser = auth()->user()->id;
        $this->users = User::whereDoesntHave('roles', function ($query) {
            $query->where('name', 'client');
        })->get();
    }

    public function render()
    {
        return view('livewire.dropdown-filter');
    }
}
```
The following is my component for the task list:
```
<?php

namespace App\Livewire;

use Livewire\WithPagination;
use Livewire\Attributes\On;
use Livewire\Component;
use App\Models\Task;
use App\Models\User;

class IndexTask extends Component
{
    use WithPagination;

    #[On('user-changed')]
    public function updatedTaskList($selectedUser)
    {
        $user = User::find($selectedUser);
        $this->tasks = Task::whereIn('id', $user->tasks()->pluck('id'))->orderBy('completed', 'ASC')->orderBy('due_date')->orderBy('name')->paginate(15);
    }

    public function render()
    {
        return view('livewire.index-task')->with([
            'tasks' => Task::whereIn('id', auth()->user()->tasks()->pluck('id'))->orderBy('completed', 'ASC')->orderBy('due_date')->orderBy('name')->paginate(15)
        ]);
    }
}
```
I can see Livewire is firing on the dropdown. The blade file for the dropdown looks like the following:
```
<div>
    @if (Auth::user()->isAdmin())
    <div class="mr-6">
        <select wire:model.live="selectedUser">
            @foreach ($users as $user)
            <option value="{{ $user->id }}">{{ $user->name }}</option>
            @endforeach
        </select>
    </div>
    @endif
</div>
```

I get the following JSON payload on the selection of a new user to filter but the actual table doesn't change:
<img width="545" alt="image" src="https://github.com/livewire/livewire/assets/13721164/4fd7fb9e-966c-4702-90ce-964549f935cf">

Any insight/help would be greatly appreciated.

**Comment by joshhanley** [comment]
@etmartinkazoo can you put a `dd('test');` in `updatedTaskList()` to make sure the method is actually being called?

**Comment by etmartinkazoo** [comment]
Hi @joshhanley - Thanks for the quick response!

Just added the `dd('test');` per your recommendation and it is firing:

<img width="692" alt="image" src="https://github.com/livewire/livewire/assets/13721164/fb08cc64-f0b5-4c7d-afff-e9ad648bf69b">

**Reply by joshhanley** [reply]
@etmartinkazoo ok great! At least the event is firing correctly. I suspect the issue might be the fact that you are passing the whole user model into the event - which I don't recommend as that will send a serialised user model to the front end. Try just passing the user ID `$this->dispatch('user-changed', $this->selectedUser->id);`.

**Reply by etmartinkazoo** [reply]
@joshhanley - I believe it is only the id from the select field which I am using an intval on and passing that in the dispatch.
```
<div>
    @if (Auth::user()->isAdmin())
    <div class="mr-6">
        <select wire:model.live="selectedUser">
            @foreach ($users as $user)
            <option value="{{ $user->id }}">{{ $user->name }}</option>
            @endforeach
        </select>
    </div>
    @endif
</div>
```

```
    public $selectedUser = null;

    public $users;

    public function updatedSelectedUser()
    {
        $this->selectedUser = intval($this->selectedUser);
        $this->dispatch('user-changed', $this->selectedUser);
    }
```

**Reply by joshhanley** [reply]
@etmartinkazoo ah ok, all good! Actually I think I just spotted the issue. You have `tasks` passed into the view and a `$tasks` public property. They will be conflicting. Try using a computed property for your tasks instead.

**Reply by etmartinkazoo** [reply]
@joshhanley - I've been trying with a computed property and am not getting anything this way either. Perhaps I'm missing something.
```
<?php

namespace App\Livewire;

use Livewire\WithPagination;
use Livewire\Attributes\On;
use Livewire\Component;
use App\Models\Task;
use App\Models\User;
use Livewire\Attributes\Computed;

class IndexTask extends Component
{
    use WithPagination;

    #[Computed]
    #[On('user-changed')]
    public function updatedTaskList($selectedUser)
    {
        $user = User::find($selectedUser);
        return Task::whereIn('id', $user->tasks()->pluck('id'))->orderBy('completed', 'ASC')->orderBy('due_date')->orderBy('name')->paginate(15);
    }

    public function render()
    {
        return view('livewire.index-task')->with([
            'tasks' => Task::whereIn('id', auth()->user()->tasks()->pluck('id'))->orderBy('completed', 'ASC')->orderBy('due_date')->orderBy('name')->paginate(15)
        ]);
    }
}
```

**Reply by etmartinkazoo** [reply]
@joshhanley - I stepped away from this as I was spinning my wheels and am going to re-approach it today for another look. I will let you know if I have a break through. It seems that pagination and mounting don't play well together. I am a novice with Livewire so I am sure my ignorance is shining on this one. I really appreciate you helping me out with this.

**Reply by etmartinkazoo** [reply]
@joshhanley - We are still stuck on this issue unfortunately. At this point we are using a non Livewire implementation of pagination until we can figure out the issue. Can't ultimately tell if this is a Livewire 3 Beta issue or something more.

**Comment by etmartinkazoo** [comment]
Hi @joshhanley - I figured it out. It ultimately hinged on making the property `protected` as opposed to `public`. The code is below. I definitely want to better understand why this works although I am happy it works :) Thank you kindly for your help.
```
<?php

namespace App\Livewire;

use Livewire\WithPagination;
use Livewire\Attributes\On;
use Livewire\Component;
use App\Models\Task;
use App\Models\User;

class IndexTask extends Component
{
    use WithPagination;

    public $selectedUser;
    protected $tasks = [];

    #[On('user-changed')]
    public function updatedIndexTask($selectedUser)
    {
        $this->resetPage();
        $user = User::find($selectedUser);
        $this->tasks = Task::whereIn('id', $user->tasks()->pluck('id'))->orderBy('completed', 'ASC')->orderBy('due_date')->orderBy('name')->paginate(15);
    }

    public function mount() {
        $this->selectedUser = auth()->user()->id;
        $this->tasks = Task::whereIn('id', auth()->user()->tasks()->pluck('id'))->orderBy('completed', 'ASC')->orderBy('due_date')->orderBy('name')->paginate(15);
    }

    public function render()
    {
        return view('livewire.index-task', [
            'tasks' => $this->tasks,
        ]);
    }
}
```
